### DISCUSSION TITLE
Postgres Full Text Search provider

### DISCUSSION THREAD

**Discussion by pokonski** [medusajs/medusa#5266] [Q&A]
Since Medusa uses Postgres, why is there no Postgres full text search plugin? Are we forced to use external SaaS  (Algolia) or self hosted search providers (MeiliSearch/ES)? 

They seem like a major overkill for small e-commerce apps and I don't see docs recommending built-in solutions that ship with the DB.

**Comment by pevey** [comment]
I think the reason is that for that you don't need a plugin, i.e., an integration with a third-party service.

You can just implement it yourself via a custom API endpoint.

**Comment by pokonski** [comment]
Yeah that's kind of my question: if we have it built in, why not use FTS as a _default_ search provider instead of making users write their own solution or be forced to rely on third-parties?

**Reply by sharif2315** [reply]
I'm wondering the same thing, why are we having to use MeiliSearch and Algolia when we should be able to use postgres FTS??

**Comment by ozayr** [comment]
Has there been any examples or progress with full text search natively in medusa via postgres ?
